org/postgresql/core/Parser.java

```diff
@@ -7,12 +7,143 @@
+                break;
+
+            case '"': // double-quotes
+                i = Parser.parseDoubleQuotes(aChars, i);
+                break;
+
+            case '-': // possibly -- style comment
+                i = Parser.parseLineComment(aChars, i);
+                break;
+
+            case '/': // possibly /* */ style comment
+                i = Parser.parseBlockComment(aChars, i);
+                break;
+
+            case '$': // possibly dollar quote start
+                i = Parser.parseDollarQuotes(aChars, i);
+                break;
+
+            case '(':
+                inParen++;
+                break;
+
+            case ')':
+                inParen--;
+                break;
+
+            case '?':
+                nativeSql.append(aChars, fragmentStart, i - fragmentStart);
+                if (i + 1 < aChars.length && aChars[i + 1] == '?') /* replace ?? with ? */
+                {
+                    nativeSql.append('?');
+                    i++; // make sure the coming ? is not treated as a bind
+                } else
+                {
+                    if (bindPositions == null)
+                        bindPositions = new ArrayList<Integer>();
+                    bindPositions.add(nativeSql.length());
+                    int bindIndex = bindPositions.size();
+                    nativeSql.append(NativeQuery.bindName(bindIndex));
+                }
+                fragmentStart = i + 1;
+                break;
+
+            case ';':
+                if (inParen == 0 && splitStatements)
+                {
+                    nativeSql.append(aChars, fragmentStart, i - fragmentStart);
+                    fragmentStart = i + 1;
+                    if (nativeSql.length() > 0)
+                    {
+                        if (nativeQueries == null)
+                            nativeQueries = new ArrayList<NativeQuery>();
+
+                        nativeQueries.add(new NativeQuery(nativeSql.toString(), toIntList(bindPositions)));
+                    }
+                    // Prepare for next query
+                    if (bindPositions != null)
+                        bindPositions.clear();
+                    nativeSql.setLength(0);
+                }
+                break;
+
+            default:
+                break;
+            }
+        }
+
+        if (fragmentStart < aChars.length)
+            nativeSql.append(aChars, fragmentStart, aChars.length - fragmentStart);
+
+        if (nativeSql.length() == 0)
+            return nativeQueries != null ? nativeQueries : Collections.<NativeQuery>emptyList();
+
+        NativeQuery lastQuery = new NativeQuery(nativeSql.toString(), toIntList(bindPositions));
+
+        if (nativeQueries == null)
+            return Collections.singletonList(lastQuery);
+
+        nativeQueries.add(lastQuery);
+        return nativeQueries;
+    }
+
+    private static int[] toIntList(List<Integer> list) {
```

Comment:
This should probably be called `toIntArray` as it is explicitly converting *from* a `List`.